diff --git a/lib/roles/ws/server-ws.c blib/roles/ws/server-ws.c
index 82e3ed9898..4e393195ad 100644
--- a/lib/roles/ws/server-ws.c
+++ b/lib/roles/ws/server-ws.c
@@ -701,6 +701,11 @@ handshake_0405(struct lws_context *context, struct lws *wsi)
 	strcpy(p, (char *)pt->serv_buf);
 	p += accept_len;
 
+#if defined(__m68k__)
+    /* Add CRLF explicitly workaround */
+    LWS_CPYAPP(p, "\x0d\x0a");
+#endif
+
 	/* we can only return the protocol header if:
 	 *  - one came in, and ... */
 	if (lws_hdr_total_length(wsi, WSI_TOKEN_PROTOCOL) &&
